tests/unittests/test_util.py

```diff
@@ -3026,3 +3026,15 @@ def test_to_version_and_back_to_str(self, version):
     )
     def test_from_str(self, str_ver, cls_ver):
         assert util.Version.from_str(str_ver) == cls_ver
+
+
+@pytest.mark.allow_dns_lookup
+class TestResolvable:
+    def test_ips_need_not_be_resolved(self):
+        """Optimization test: dns resolution may timeout during early boot, and
+        often the urls being checked use IP addresses rather than dns names.
+        Therefore, the fast path checks if the address contains an IP and exits
+        early if the path is a valid IP.
+        """
+        assert util.is_resolvable("http://169.254.169.254/")
+        assert util.is_resolvable("http://[fd00:ec2::254]/")
```

Comment:
I think we should be checking that `socket.getaddrinfo` isn't called here.

Reply:
It gets used higher up in the function, but I can mock the global variable to skip that code.